sstables/mp_row_consumer.hh

```diff
@@ -974,13 +978,9 @@ public:
 
     // See the RowConsumer concept
     void push_ready_fragments() {
-        if (!_mf_filter || _mf_filter->out_of_range()) {
```

Comment:
What about _mf_filter updates in 'setup_for_partition' and in 'consume_row_end'? Is it impossible for those places to put _mf_filter out of range?

Reply:
I don't think so. Can you point to the place where this may happen?

Reply:
https://github.com/scylladb/scylla/blob/ef6edff673fe40f32d0a2fbc318778cdf7c62241/sstables/mp_row_consumer.hh#L1015

https://github.com/scylladb/scylla/blob/ef6edff673fe40f32d0a2fbc318778cdf7c62241/sstables/mp_row_consumer.hh#L1289

Reply:
Constructing a filter doesn't make it enter out-of-range. 

In the second case, we're consuming a static row. That could make the filter go out of range only if it selects nothing in the partition, not even the static row. I didn't think that's an interesting case to optimize. Nothing wrong would happen if we don't catch such out of range condition here, it will be caught when the next clustered fragment is consumed, or partition end is reached. 

Note that before the patch we'd usually return `proceed::yes` here, so `push_ready_fragments()` wouldn't have a chance to intervene anyway.

Reply:
ok